lib/semian/atomic_enum.rb

```diff
@@ -0,0 +1,39 @@
+require 'forwardable'
+
+module Semian
+  class AtomicEnum #:nodoc:
+    extend Forwardable
+
+    def_delegators :@integer, :execute_atomically, :transaction, :shared?, :destroy
+    private :shared?
+
+    def initialize(symbol_list, **_options)
```

Comment:
I wanted to keep the interface the same. Otherwise things like `name: 'blah'`, etc in the initializer will throw errors.